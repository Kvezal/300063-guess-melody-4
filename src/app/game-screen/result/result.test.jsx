import React from "react";
import render from "react-test-renderer";

import Result from "./result";


describe(`Result`, () => {
  test(`should render success component`, () => {
    const tree = render
      .create(
          <Result type="success" rightAnswers={10} mistakes={0}/>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test(`should render failed component`, () => {
    const tree = render
      .create(
          <Result type="failed" rightAnswers={0} mistakes={0}/>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});

