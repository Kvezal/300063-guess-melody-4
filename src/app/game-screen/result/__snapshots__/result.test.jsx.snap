// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Result should render failed component 1`] = `
<section
  className="result"
>
  <div
    className="result__logo"
  >
    <img
      alt="Угадай мелодию"
      height="83"
      src="img/melody-logo.png"
      width="186"
    />
  </div>
  <h2
    className="result__title"
  >
    Какая жалость!
  </h2>
  <p
    className="result__total result__total--fail"
  >
    У вас закончились все попытки. Ничего, повезёт в следующий раз!
  </p>
  <button
    className="replay"
    type="button"
  >
    Попробовать ещё раз
  </button>
</section>
`;

exports[`Result should render success component 1`] = `
<section
  className="result"
>
  <div
    className="result__logo"
  >
    <img
      alt="Угадай мелодию"
      height="83"
      src="img/melody-logo.png"
      width="186"
    />
  </div>
  <h2
    className="result__title"
  >
    Вы настоящий меломан!
  </h2>
  <p
    className="result__total"
  >
    Вы ответили правильно на 
    10
     вопросов и совершили 
    0
     ошибки
  </p>
  <button
    className="replay"
    type="button"
  >
    Попробовать ещё раз
  </button>
</section>
`;
